pyct/report.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+# Import and print the library version and filesystem location for each Python package or shell command specified
+#
+# bash usage: $ autover numpy pandas python conda
+# python usage: >>> import autover ; autover("numpy","pandas","python","conda")
+
+from __future__ import print_function
+import os.path, importlib, subprocess, platform, sys
+
+def report(*packages):
+    """Import and print location and version information for specified Python packages"""
+    accepted_commands = ['python','conda']
+    for package in packages:
+        loc = "not installed in this environment"
+        ver = "unknown"
+
+        try:
+            module = importlib.import_module(package)
+            loc =  os.path.dirname(module.__file__)
+
+            try:
+                ver = str(module.__version__)
+            except Exception:
+                pass
+        
+        except (ImportError, ModuleNotFoundError):
+            if package in accepted_commands:
+                try:
+                    # See if there is a command by that name and check its --version if so
+                    loc = subprocess.check_output(['which', package]).decode().strip()
+                    out = ""
+                    try:
+                        out = subprocess.check_output([package, '--version'], stderr=subprocess.STDOUT)
```

Comment:
I think Chris's comment on this PR amounts to suggesting adding "shell=True" to this command, which I agree would improve it if it works as desired (making it launch a shell first in which to run the command).  To address his other comment, maybe try where.exe if which fails; see [this SO post](https://superuser.com/questions/207707/what-is-windows-equivalent-of-the-which-command-in-unix-is-there-an-equivale).

Reply:
> suggesting adding "shell=True" to this command, which I agree would improve it if it works as desired (making it launch a shell first in which to run the command)

Yes, usually you do not want `shell=True`, but if you are trying to report what someone is likely running when they are having a problem, maybe you do.

I am not actually suggesting to switch, because I don't understand it well enough to be sure. My actual suggestion would be to try it one way or the other and see how it goes (being prepared for some misreporting from the command in practice, and perhaps being ready to change if so). Because I don't know the answer. But if someone actually has experience and knows what to do (I feel like a conda person would know, for example), that would be great. 

Just to repeat the pain I was having: while using conda (and switching/stacking/manipulating environments via conda a lot), I would frequently be left in a situation where typing `conda` at the shell would do something different from `conda` without the shell. You can imagine how that happened, between env stacking, shell hashing, changes to what "conda" actually was (function, command), etc. But I believe that is better/has stabilized now.  

> To address his other comment, maybe try where.exe if which fails; see this SO post.

I actually linked to where I did that already:

![Screenshot from 2020-05-01 08-06-24](https://user-images.githubusercontent.com/1929/80789007-ae082480-8b82-11ea-93a6-08c2eafcd71c.png)

I also linked to how to add tests that the commands at least run without failing: 

![Screenshot from 2020-05-01 08-22-53](https://user-images.githubusercontent.com/1929/80789808-04766280-8b85-11ea-892f-b90ed7a9645f.png)

(You could check output also...)

Reply:
Perfect, thanks!  Kenneth, just need to hit it home and actually capture and apply Chris's work here!

Reply:
After running pyct with `shell=True` I've been running into issues with conda and python not being shown as installed, so this may not be the way to go.

Reply:
I've also tested on Windows and the `where.exe` update is working.

Reply:
Ok, sounds like we're done! Thanks.

Reply:
> I've been running into issues with conda and python not being shown as installed

Thanks for checking about `shell=True`; I agree we should go with `False` if that seems to be working. 

While I understand why `shell=False` is usually what you want to be using for running stuff, this case seems like one of the rare ones where `shell=True` makes sense. Oh well!

Incidentally, I found the following link while trying to understand a bit better, but I soon closed the tab and swept this topic back into the nightmare category of things I don't want to know about. I'd just hoped we could figure it out for sure one time and then never think about it again, but I see that is probably not possible.

https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then.

I see that conda activate/deactivate does `hash -r` now (which I am reasonably - but not completely - sure it did not do in the distant past), so I think that will help.

Reply:
Well, I did have a bit of a look. I see at least two changes in conda that make me worry less: 
  * When you are in some environment, the base environment's bin folder is no longer on the path (which I am reasonably but not completely certain used to be the case). So if pyct is installed in the base, but not the current env, running `pyct version` will no longer report the wrong things (i.e. contents of base env) - it will fail to run. That's good.
  * conda activate/deactivate does `hash -r` now (which I am reasonably but not completely certain used to be the case). So when someone types `x` in bash, it's more likely to be the same `x` found by pyct (which bypasses the shell).

Again though, don't really know what I am talking about here :)